.github/workflows/deploy-react-storybook-prod.yml

```diff
@@ -9,8 +9,46 @@ env:
   AWS_S3_BUCKET_NAME: storybook-react-design-system-blocks-gov-ie
 
 jobs:
+  determine-affected:
+    runs-on: ubuntu-latest
+    outputs:
+      react_affected: ${{ steps.check-affected.outputs.react_affected }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.x"
+
+      - name: Install dependencies
+        run: |
+          corepack enable
+          pnpm install --frozen-lockfile --ignore-scripts
+
+      - name: Determine affected projects
+        id: affected
+        run: |
+          pnpm nx show projects --affected --base=origin/main~1 --head=HEAD > affected.txt
+          cat affected.txt
+
+      - name: Set environment output if HTML package is affected
+        id: check-affected
+        run: |
+          if grep -q "@govie-react/storybook" affected.txt; then
+            echo "react_affected=true" >> $GITHUB_OUTPUT
+          else
+            echo "react_affected=false" >> $GITHUB_OUTPUT
+          fi
+
   build:
     runs-on: ubuntu-latest
+    needs: determine-affected
+    # Run if html is affected or manually triggered
+    if: ${{ needs.determine-affected.outputs.react_affected == 'true' || github.event_name == 'workflow_dispatch' }}
```

Comment:
Given that this check is here, we might need manual trigger here :D

Reply:
thanks, forgot about it